### DISCUSSION TITLE
Swap icons with rotate effect problem

### DISCUSSION THREAD

**Discussion by lorenzo-mm** [saadeghi/daisyui#1848] [Q&A] [answered]
Hello, I'm trying to use the night mode icon but I don't know why it doesn't work out as in the example, I'll take a screenshot to see it better:
[image](https://github.com/saadeghi/daisyui/assets/94995682/97120355-8bbf-4466-85b0-63cd930e8c8e)

I don't know why the sun stays next to the moon instead of one on top of the other and when I change the mode the icon changes as in the example => https://daisyui.com/components/swap/#swap-icons-with-rotate-effect
And how can I remove the checkbox?

Thanks!

**Comment by saadeghi** [comment]
guessing by your screenshot, no style is applied here.  
- Are you using a really old version of daisyUI?
- Are you sure the path to all your template files and components are addressed in `tailwind.config.js` [in `content` array](https://tailwindcss.com/docs/content-configuration)?

**Reply by lorenzo-mm** [reply]
Yes I am using the latest version of DaisyUI

these are my routes in tailwind.config.js
```
  content: [
    './src/**/*.{js,ts,jsx,tsx}',
    './components/**/*.{js,ts,jsx,tsx}'
  ],
```

**Reply by saadeghi** [reply]
Can you make a minimal reproduction link for me to see the issue?  
Can be either a [Tailwind Play link](https://daisyui.com/tailwindplay/), CodeSandBox or StackBlitz, etc.  

Because the CSS has no issues and you can see it's working here: https://play.tailwindcss.com/uFgFR4ZJHk  
So it's probably a config issue in your project.

**Reply by lorenzo-mm** [reply]
I have managed to make them put one on top of the other and that when clicking the icon changes, 
what does not work is the effect to do when it changes from one to another, Is it mandatory to put the checkbox?

I try to put the reproduction

**Reply by saadeghi** [reply]
>Is it mandatory to put the checkbox?

The checkbox status (checked/uncheck) controls which icon should be visible.  
If you want to control it using JS instead, you can add/remove `swap-active` class instead of using the checkbox like the last code example in this page: https://daisyui.com/components/swap/#activate-using-class-name-instead-of-checkbox

**Reply by lorenzo-mm** [reply]
I have seen from the link that you have given me, that if I remove the status of the checkbox it does not have the effect of rotating the icon,
My question is, can I use css customs and tailwimd at the same time or am I forced to do the effect with tailwind?

**Reply by saadeghi** [reply]
>if I remove the status of the checkbox it does not have the effect of rotating the icon  

Not true. If you use `swap-rotate` class it will have the rotate effect when it's changing. Either changing using a hidden checkbox or changing by adding/removing `swap-active` class.

**Reply by lorenzo-mm** [reply]
ok i see, i added the swap-rotate effects and now the rotation works, but when i click it and the sun rises it shows for a moment and then it disappearsok i see, i added the swap-rotate effects and now the rotation works, but when I click on it and the sun rises it appears for a moment and then it disappears

**Reply by lorenzo-mm** [reply]
```
const sun = (
  <svg
    className='swap-on fill-current w-10 h-10'
    xmlns='http://www.w3.org/2000/svg'
    viewBox='0 0 24 24'
  >
    <path
      d='M5.64,17l-.71.71a1,1,0,0,0,0,1.41,1,1,0,0,0,1.41,0l.71-.71A1,1,0,0,0,5.64,17ZM5,12a1,1,0,0,0-1-1H3a1,1,0,0,0,0,2H4A1,1,0,0,0,5,12Zm7-7a1,1,0,0,0,1-1V3a1,1,0,0,0-2,0V4A1,1,0,0,0,12,5ZM5.64,7.05a1,1,0,0,0,.7.29,1,1,0,0,0,.71-.29,1,1,0,0,0,0-1.41l-.71-.71A1,1,0,0,0,4.93,6.34Zm12,.29a1,1,0,0,0,.7-.29l.71-.71a1,1,0,1,0-1.41-1.41L17,5.64a1,1,0,0,0,0,1.41A1,1,0,0,0,17.66,7.34ZM21,11H20a1,1,0,0,0,0,2h1a1,1,0,0,0,0-2Zm-9,8a1,1,0,0,0-1,1v1a1,1,0,0,0,2,0V20A1,1,0,0,0,12,19ZM18.36,17A1,1,0,0,0,17,18.36l.71.71a1,1,0,0,0,1.41,0,1,1,0,0,0,0-1.41ZM12,6.5A5.5,5.5,0,1,0,17.5,12,5.51,5.51,0,0,0,12,6.5Zm0,9A3.5,3.5,0,1,1,15.5,12,3.5,3.5,0,0,1,12,15.5Z'
    />
  </svg>
)

const moon = (
  <svg
    className='swap-off fill-current w-10 h-10'
    xmlns='http://www.w3.org/2000/svg'
    viewBox='0 0 24 24'
  >
    <path
      d='M21.64,13a1,1,0,0,0-1.05-.14,8.05,8.05,0,0,1-3.37.73A8.15,8.15,0,0,1,9.08,5.49a8.59,8.59,0,0,1,.25-2A1,1,0,0,0,8,2.36,10.14,10.14,0,1,0,22,14.05,1,1,0,0,0,21.64,13Zm-9.5,6.69A8.14,8.14,0,0,1,7.08,5.22v.27A10.15,10.15,0,0,0,17.22,15.63a9.79,9.79,0,0,0,2.1-.22A8.11,8.11,0,0,1,12.14,19.73Z'
    />
  </svg>
)

export default function BtnDarkMode () {
  const [isDarkMode, setIsDarkMode] = useState(false)

  const handleToggleTheme = () => {
    setIsDarkMode(!isDarkMode)
    document.body.classList.toggle('dark-mode')
  }

  return (
    button className='swap swap-rotate' onClick={handleToggleTheme}
      {isDarkMode
        ? (
          img src={sun} alt='' 
          )
        : (
          img src={moon} alt='' 
          )}
      {isDarkMode ? moon : sun}
    </button>
  )
}
```
this is my code

**Reply by saadeghi** [reply, marked as answer]
@lorenzo-mm My friend, you're walking the wrong path. You're just showing and hiding the image using JS so you can't expect to see the CSS effect.
Here's an example for you: https://stackblitz.com/edit/daisyui-react-vite-p9aqsp?file=src%2FApp.jsx

**Reply by lorenzo-mm** [reply]
you are right, but with the moon you can see the effect, but I understand what you mean, now it works as it should, thank you very much for the help and your time @saadeghi
